File: frontend/src/types/characterDashboard.ts
export interface CharacterDashboardItem {
  character: string;
  pinyin: string;
  meaning: string;
  status: string;
  successful_recognitions: number;
  failed_recognitions: number;
  correct_pinyin_count: number;
  times_seen: number;
  last_practised_at: string | null;
  last_seen_at: string | null;
}

export interface CharacterDashboardResponse {
  items: CharacterDashboardItem[];
  total: number;
  status_counts: Record<string, number>;
}

export interface CharacterDashboardFilters {
  status?: string | null;
  search?: string | null;
}

File: backend/src/chinese_learning/presentation/rest/schemas/character_dashboard.py
"""Schemas for the character dashboard API."""

from pydantic import BaseModel


class CharacterDashboardItemSchema(BaseModel):
    character: str
    pinyin: str
    meaning: str
    status: str
    successful_recognitions: int
    failed_recognitions: int
    correct_pinyin_count: int
    times_seen: int
    last_practised_at: str | None = None
    last_seen_at: str | None = None


class CharacterDashboardResponse(BaseModel):
    items: list[CharacterDashboardItemSchema]
    total: int
    status_counts: dict[str, int]
